### PR TITLE
dryrun stats observable

### PR DESCRIPTION
**Pull request by jloftin-nv**
# Description

dryrun=True builds a full DryRunTracker (coverage, unsupported ops, engine count) and only prints it. There is no supported way to read those stats for CI or sweeps without monkeypatching.

Mark dryrun_stats_display with @observable() so callers can register on observers.pre / post and capture the tracker without a new public API.

Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.

- New feature (non-breaking change which adds functionality)

# Checklist:

- [X ] My code follows the style guidelines of this project (You can use the linters)
- [ X] I have performed a self-review of my own code
- [ X] I have commented my code, particularly in hard-to-understand areas and hacks
- [X ] I have made corresponding changes to the documentation
- [ X] I have added tests to verify my fix or my feature
- [ X] New and existing unit tests pass locally with my changes
- [ X] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: tests/py/dynamo/runtime/test_dryrun_stats_observable.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+        model = Add().cuda().eval()
+        x = torch.randn(2, 3, device="cuda")
+        trackers = []
+
+        def capture(ctx: ObserveContext) -> None:
+            trackers.append(ctx.args[0])
+
+        with dryrun_stats_display.observers.pre.add(capture):
+            torch_tensorrt.dynamo.compile(
```
Comment (micwill755) [inline tests/py/dynamo/runtime/test_dryrun_stats_observable.py:27]: dynamo.compile() needs an ExportedProgram, not an nn.Module. Export first with torch.export.export(model, (x,)), then pass that into compile. 

Also drop enabled_precisions={torch.float32}, enabled_precisions is deprecated as of TRT 11. Dynamo uses dtypes on the model and input so this FP32 test needs no precision argument.

**Review by micwill755 — CHANGES_REQUESTED** [review]
Request changes, for these:

1) Docs missing. Checklist says docs were updated; they were not. Add the callback recipe to dryrun docs / observer docs.

2) Hook does not always run. dryrun_stats_display is skipped on some compile_module early returns including num_supported_ops == 0 even with dryrun=True. Coverage sweeps of unsupported models would get no tracker. Call display or emit the tracker on those paths too.

**Comment by jloftin-nv** [comment]
@micwill755 All should be addressed. LMK if it looks good
